tests/test_examples.py

```diff
@@ -1050,3 +1054,37 @@ def test_example_get_current(self):
                 -1.36284423e07,
             ],
         )
+
+
+@pytest.mark.unit
+def test_single_coil_optimization():
+    """Test that single coil (not coilset) optimization works."""
+    # testing that the objectives work and that the optimization framework
+    # works when a single coil is passed in.
+
+    opt = Optimizer("fmintr")
+    coil = FourierRZCoil()
+    # length
+    target = 60
+    obj = ObjectiveFunction(CoilLength(coil, target=target))
+    opt.optimize([coil], obj)
+    np.testing.assert_allclose(coil.compute("length")["length"], target)
+
+    # curvature
+    grid = LinearGrid(N=10)
+    target = 2 * coil.compute("curvature", grid=grid)["curvature"]
+    obj = ObjectiveFunction(CoilCurvature(coil, target=target, grid=grid))
+    opt.optimize([coil], obj, maxiter=300)
+    np.testing.assert_allclose(
+        coil.compute("curvature", grid=grid)["curvature"], target, rtol=1e-3
+    )
+
+    # torsion
+    # this test is not working very well... unsure of why
```

Comment:
constant torsion and curvature is a helix, but if the pitch isn't right then it might not be periodic, so that might be the issue. 

A simpler test would be to initialize with some non-planar shape and target zero torsion, which should make it planar

Reply:
True, thanks